File: src/main/java/com/skyspace33/service/RuleService.java
package com.skyspace33.service;

import java.util.List;

import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;
import org.springframework.data.jpa.domain.Specification;
import org.springframework.http.ResponseEntity;

import com.skyspace33.domain.Rule;
import com.skyspace33.dto.RuleDTO;
import com.skyspace33.dto.RuleSearchDTO;
import com.skyspace33.dto.RulePageDTO;
import com.skyspace33.dto.RuleConvertCriteriaDTO;
import com.skyspace33.service.GenericService;
import com.skyspace33.dto.common.RequestDTO;
import com.skyspace33.dto.common.ResultDTO;
import java.util.List;
import java.util.Optional;





public interface RuleService extends GenericService<Rule, Integer> {

	List<Rule> findAll();

	ResultDTO addRule(RuleDTO ruleDTO, RequestDTO requestDTO);

	ResultDTO updateRule(RuleDTO ruleDTO, RequestDTO requestDTO);

    Page<Rule> getAllRules(Pageable pageable);

    Page<Rule> getAllRules(Specification<Rule> spec, Pageable pageable);

	ResponseEntity<RulePageDTO> getRules(RuleSearchDTO ruleSearchDTO);
	
	List<RuleDTO> convertRulesToRuleDTOs(List<Rule> rules, RuleConvertCriteriaDTO convertCriteria);

	RuleDTO getRuleDTOById(Integer ruleId);







}

File: skyspace33_ui/src/services/RuleService.js
import http from "../http-common"; 

class RuleService {
  getAllRules(searchDTO) {
    console.log(searchDTO)
    return this.getRequest(`/rule/rules`, searchDTO);
  }

  get(ruleId) {
    return this.getRequest(`/rule/${ruleId}`, null);
  }

  findByField(matchData) {
    return this.getRequest(`/rule?field=${matchData}`, null);
  }

  addRule(data) {
    return http.post("/rule/addRule", data);
  }

  update(data) {
  	return http.post("/rule/updateRule", data);
  }
  
  uploadImage(data,ruleId) {
  	return http.postForm("/rule/uploadImage/"+ruleId, data);
  }




	postRequest(url, data) {
		return http.post(url, data);
      };

	getRequest(url, params) {
        return http.get(url, {
        	params: params
        });
    };

}

export default new RuleService();
